lib/connections/service.go

```diff
@@ -394,8 +377,8 @@ func (s *Service) connect() {
 
 				prio := dialerFactory.Priority()
 
-				if priorityKnown && prio >= ct.internalConn.priority {
-					l.Debugf("Not dialing using %s as priority is less than current connection (%d >= %d)", dialerFactory, dialerFactory.Priority(), ct.internalConn.priority)
+				if ok && prio >= ct.Priority() {
```

Comment:
The `ok` needs a proper name. This panics because ok was used for something else since its creation.